tests/test_check_parallel.py

```diff
@@ -231,6 +231,24 @@ def test_worker_check_single_file_no_checkers(self) -> None:
         assert stats.statement == 18
         assert stats.warning == 0
 
+    def test_linter_with_unpickleable_plugins_is_pickleable(self) -> None:
+        """The linter needs to be pickle-able in order to be passed between workers"""
+        linter = PyLinter(reporter=Reporter())
+        # We load an extension that we know is not pickle-safe
+        linter.load_plugin_modules(["pylint.extensions.overlapping_exceptions"])
+        try:
+            dill.dumps(linter)
+            assert False, "Plugins loaded were pickle-safe! This test needs altering"
+        except (KeyError, TypeError, dill.PickleError):
```

Comment:
When testing it locally, this throws a `NotImplementedError` likely because of `wrapt`. Adding the exception should be enough. Just interesting that the CI workflow is green 🤔

> If pickle.dump() or pickle.dumps() is used on an instance of the ObjectProxy class, a NotImplementedError exception is raised, with a message indicating that the object proxy must implement the reduce_ex() method. This is in place of the default TypeError exception with message indicating a pickle error.

https://wrapt.readthedocs.io/en/latest/changes.html#version-1-11-0

Reply:
This only seems to be an issue if astroid is installed in editable mode. Not exactly sure why though. Adding `NotImplementedError` does resolve it.